### PR TITLE
Add logic to schedules

### PR DESCRIPTION
**Pull request by tlunter**
@nahiluhmot @djosborne

### CODE REVIEW

File: app/cappy/controllers/base.rb [hunk 1]
```diff
@@ -10,9 +10,9 @@ class Base < Sinatra::Base
 
       error do |err|
         case err
-        when Errors::MalformedRequestError, Errors::BadDeviceOptions, Errors::BadStateOptions
+        when Errors::MalformedRequestError, Errors::BadOptions
           status 400
-        when Errors::NoSuchDevice
+        when Errors::NoSuchObject
```
Comment (nahiluhmot) [inline app/cappy/controllers/base.rb]: Nice error consolidating.

File: app/cappy/services/base.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Cappy
+  module Services
+    # Simple base module for utility methods
+    module Base
+      module_function
+
+      def included(base)
+        base.extend(self)
```
Comment (nahiluhmot) [inline app/cappy/services/base.rb]: `module_functions` + `mixins` :)

File: app/cappy/services/schedules.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    module Schedules
+      include Base
+
+      module_function
+
+      def list(device_id = nil)
+        if device_id.nil?
+          Models::Schedule.all.map(&:as_json)
+        else
+          device = Services::Devices.get_device(device_id)
```
Comment (nahiluhmot) [inline app/cappy/services/schedules.rb]: Why not a simple `where` query? Would save one trip to the db.

Comment (nahiluhmot) [inline app/cappy/services/schedules.rb]: Also, perhaps these should be separate methods, such as `list` and `for_device`.

File: spec/app/cappy/controllers/schedules_spec.rb [hunk 4]
```diff
@@ -0,0 +1,147 @@
+require 'spec_helper'
+require 'pry'
```
Comment (nahiluhmot) [inline spec/app/cappy/controllers/schedules_spec.rb]: Shouldn't this be included via `Bundler.require(:test)`?

**Comment by nahiluhmot** [comment]
Looks good for the most part, just a few minor nitpicks.

File: spec/app/cappy/controllers/schedules_spec.rb (hunk 4 above)
Comment (tlunter) [inline spec/app/cappy/controllers/schedules_spec.rb]: True, probably is. Just did it out of habit.  Was for debug anyway, so I'll remove it.

File: app/cappy/services/schedules.rb (hunk 3 above)
Comment (tlunter) [inline app/cappy/services/schedules.rb]: `device_id` in this case isn't the primary key, but the String `device_id` off `device`, and `device_id` on `schedule` is the primary key of `device`.  It's an awkward query I think.

Comment (tlunter) [inline app/cappy/services/schedules.rb]: Just kidding, I figured it out.

Comment (nahiluhmot) [inline app/cappy/services/schedules.rb]: Ah, good point. :+1:

**Pull request merged**